### PR TITLE
Fix dead links

### PR DESCRIPTION
**Pull request by bchen1116**
Fixing dead links in documentation

Docs [here](https://feature-labs-inc-evalml--3358.com.readthedocs.build/en/3358/start.html)

### CODE REVIEW

File: docs/source/release_notes.rst [hunk 1]
```diff
@@ -8,6 +8,7 @@
         * Fixed ``get_pipelines`` to reset pipeline threshold for binary cases :pr:`3360`
     * Changes
     * Documentation Changes
+        * Fixed documentation links :pr:`3358`
```
Comment (eccabay) [inline docs/source/release_notes.rst:11]: Mega nit: could this be slightly more explicit? (i.e. "fixed broken documentation links" or "fixed documentation links to point to correct pages", or something like that)

**Review by eccabay — APPROVED** [review]
LGTM, thanks for taking care of this!

File: docs/source/release_notes.rst (hunk 1 above)
Comment (bchen1116) [reply docs/source/release_notes.rst:11]: Good call, can do!

File: docs/source/user_guide/automl.ipynb [hunk 2]
```diff
@@ -417,9 +417,9 @@
     "\n",
     "AutoMLSearch will automatically fit the best pipeline on the entire training data. It also provides an easy API for training and scoring other pipelines.\n",
     "\n",
-    "If you'd like to train one or more pipelines on the entire training data, you can use the `train_pipelines`method\n",
+    "If you'd like to train one or more pipelines on the entire training data, you can use the `train_pipelines` method.\n",
     "\n",
-    "Similarly, if you'd like to score one or more pipelines on a particular dataset, you can use the `train_pipelines`method\n"
+    "Similarly, if you'd like to score one or more pipelines on a particular dataset, you can use the `score_pipelines` method.\n"
```
Comment (freddyaboulton) [inline docs/source/user_guide/automl.ipynb:422]: Good catch! 👀

**Review by freddyaboulton — APPROVED** [review]
Thank you @bchen1116 !

**Pull request merged**